### DISCUSSION TITLE
set state of store from another store

### DISCUSSION THREAD

**Discussion by bouaik** [pmndrs/zustand#1562] [Q&A] [answered]
i'm trying to set  the state of aStore from bStore like this, but not working, any ideas on how can i implement this

example

bStore.js
```javascript
const bear = // from API response
const {bears} = useAStore.getState()

// 1st method not working
set({
bears: [...bears, bear]
})

// 2nd method has an error of bears not iterable
set(state => {
  return {
       bears: [...state.bears, bear]
    }
  }
)
```

Thanks in advance

**Comment by dai-shi** [comment, marked as answer]
Can you create full (non-working) code in https://codesandbox.io/s/react-new ?

**Reply by bouaik** [reply]
Hi @dai-shi, i'm trying to implement something like this https://codesandbox.io/s/jovial-ioana-0y9rri

- bears and increment bears are in aStore.js
- increment the same bears from aStore by two in bStore.js

Thanks

**Reply by dai-shi** [reply]
Here you go: https://codesandbox.io/s/stupefied-cdn-u8jph1?file=/src/store/bStore.js

**Reply by bouaik** [reply]
Thank you so much
